File: examples/python/dynamic_scheme.py
import material_color_py as mc


def main() -> None:
    source = mc.Argb.from_rgb(66, 133, 244)
    scheme = mc.DynamicScheme(source.value, mc.Variant.TONAL_SPOT, False, 0.0)

    primary = scheme.color(mc.DynamicColorRole.PRIMARY)
    on_primary = scheme.color(mc.DynamicColorRole.ON_PRIMARY)
    container = scheme.color(mc.DynamicColorRole.PRIMARY_CONTAINER)

    print(f"material_color_py {mc.__version__}")
    print(f"source=0x{source.value:08x}")
    print(f"primary=0x{primary:08x}")
    print(f"on_primary=0x{on_primary:08x}")
    print(f"primary_container=0x{container:08x}")

    assert mc.__version__ == "1.0.0"
    assert primary == 0xFF445E91
    assert on_primary == 0xFFFFFFFF
    assert container == 0xFFD8E2FF


if __name__ == "__main__":
    main()

File: examples/rust/dynamic_scheme.rs
use material_color::VERSION;
use material_color::argb::Argb;
use material_color::dynamic::{DynamicColorRole, DynamicScheme, Variant};

fn main() {
    let source = Argb::from_rgb(66, 133, 244);
    let scheme = DynamicScheme::from_argb(source, Variant::TonalSpot, false, 0.0);

    let primary = scheme.color(DynamicColorRole::Primary);
    let on_primary = scheme.color(DynamicColorRole::OnPrimary);
    let container = scheme.color(DynamicColorRole::PrimaryContainer);

    println!("material_color {VERSION}");
    println!("source=0x{:08x}", source.to_u32());
    println!("primary=0x{:08x}", primary.to_u32());
    println!("on_primary=0x{:08x}", on_primary.to_u32());
    println!("primary_container=0x{:08x}", container.to_u32());

    assert_eq!(VERSION, "1.0.0");
    assert_eq!(primary.to_u32(), 0xff44_5e91);
    assert_eq!(on_primary.to_u32(), 0xffff_ffff);
    assert_eq!(container.to_u32(), 0xffd8_e2ff);
}
